packages/dynamite/dynamite/lib/src/helpers/built_value.dart

```diff
@@ -1,65 +1,73 @@
 import 'package:built_collection/built_collection.dart';
 import 'package:code_builder/code_builder.dart';
 import 'package:dynamite/src/helpers/dart_helpers.dart';
+import 'package:dynamite/src/type_result/type_result.dart';
+
+const interfaceSuffix = 'Interface';
 
 Spec buildBuiltClass(
-  final String className,
-  final BuiltList<Method> methods, {
+  final String className, {
+  final BuiltList<Method>? methods,
   final Iterable<String>? defaults,
+  final Iterable<TypeResultObject>? interfaces,
   final bool customSerializer = false,
-}) =>
-    Class(
-      (final b) {
-        b
-          ..name = className
-          ..abstract = true
-          ..implements.add(
-            refer(
-              'Built<$className, ${className}Builder>',
-            ),
-          )
-          ..constructors.addAll([
-            builtValueConstructor(className),
-            hiddenConstructor,
-            fromJsonConstructor,
-          ])
-          ..methods.addAll([
-            toJsonMethod,
-            ...methods,
-            buildSerializer(className, isCustom: customSerializer),
-          ]);
+}) {
+  assert(interfaces == null || methods == null, 'Either provide an interface or methods.');
```

Comment:
```suggestion
  assert((interfaces == null) != (methods == null), 'Either provide an interface or methods.');
```

I believe this should be XOR? Otherwise the wording is confusing.

Reply:
In theory a normal or should also be possible but I don't see why we should allow this.
I'll change it to an XOR accidentally mess anything up in the future.